Mathlib/Topology/Separation.lean

```diff
@@ -1440,6 +1440,109 @@ theorem Set.Finite.t2_separation [T2Space X] {s : Set X} (hs : s.Finite) :
   s.pairwiseDisjoint_nhds.exists_mem_filter_basis hs nhds_basis_opens
 #align set.finite.t2_separation Set.Finite.t2_separation
 
+/-- In a `T2Space X`, for a compact set `t` and a point `x` outside `t`, there are open sets `U`,
+`V` that separate `t` and `x`.-/
+lemma separation_of_isCompact_not_mem {X : Type u_1} [TopologicalSpace X] [T2Space X] {x : X}
+    {t : Set X} (H1 : IsCompact t) (H2 : x ∉ t) :
+    ∃ (U : Set X), ∃ (V : Set X), IsOpen U ∧ IsOpen V ∧ x ∈ U ∧ t ⊆ V ∧ Disjoint U V := by
+  obtain ⟨v, hv⟩ := Filter.disjoint_iff.mp ((IsCompact.disjoint_nhdsSet_left H1).mpr
+    (fun (y : X) (p : y ∈ t) => disjoint_nhds_nhds.mpr (ne_of_mem_of_not_mem p H2)))
+  obtain ⟨u, hu⟩ := hv.2
+  obtain ⟨V, hV⟩ := mem_nhdsSet_iff_exists.mp hv.1
+  obtain ⟨U, hU⟩ := mem_nhds_iff.mp hu.1
+  use U
+  use V
+  constructor
+  exact hU.2.1
+  constructor
+  exact hV.1
+  constructor
+  exact hU.2.2
+  constructor
+  exact hV.2.1
+  exact Set.disjoint_of_subset hU.1 hV.2.2 (Disjoint.symm hu.2)
+
+/-- In a `T2Space X`, for a compact set `t` and a point `x` outside `t`, there are neighbourhoods
+`U`, `V` that separate `t` and `x`.-/
+lemma separation_of_isCompact_not_mem' {X : Type u_1} [TopologicalSpace X] [T2Space X] {x : X}
+    {t : Set X} (H1 : IsCompact t) (H2 : x ∉ t) :
+    ∃ U ∈ nhds x,  ∃ V ∈ nhdsSet t, Disjoint U V := by
+  obtain ⟨U, hU⟩ := separation_of_isCompact_not_mem H1 H2
+  obtain ⟨V, hV⟩ := hU
+  use U
+  constructor
+  · rw [mem_nhds_iff]
+    use U
+    constructor
+    exact subset_rfl
+    constructor
+    exact hV.1
+    exact hV.2.2.1
+  · use V
+    constructor
+    · apply (IsOpen.mem_nhdsSet hV.2.1).mpr hV.2.2.2.1
+    · exact hV.2.2.2.2
+
+/-- In a `T2Space X`, for disjoint compact sets `s` and `t`, there are neighbourhoods `U`, `V` that
+separate `s` and `t`.-/
+lemma separation_of_isCompact_isCompact_disjoint {X : Type u_1} [TopologicalSpace X] [T2Space X]
```

Comment:
This is `SeparatedNhds.of_isCompact_isCompact`

Reply:
I remove separation_of_isCompact_isCompact_disjoint and prove `t2_separation_IsOpen_IsCompact_closure_IsClosed_subset` after `SeparatedNhds.of_isCompact_isCompact`.